src/Ranges/CanonicalRange.php

```diff
@@ -50,4 +50,18 @@ private function canonicalizeUpperBoundary(): void
             $this->toBound = ')';
         }
     }
+
+    public function fromBound(): string
+    {
+        $this->canonicalizeLowerBoundary();
```

Comment:
are you sure that we need to call this functions in getters? they called once in constructor, isn't it enough?

Reply:
i guess we don't need getters in this class at all, because it extends Range class that already has these getters

Reply:
> they called once in constructor, isn't it enough?

i think so too! 
I'm sorry but I overlooked that `canonicalizeBoundaries()` is called in the constructor.
I'll fix it right now.